src/services/jsTyping.ts

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the Apache License, Version 2.0. 
+// See LICENSE.txt in the project root for complete license information.
+
+/// <reference path='services.ts' />
+
+/* @internal */
+namespace ts.JsTyping {
+
+    interface HostType {
+        directoryExists: (path: string) => boolean;
+        fileExists: (fileName: string) => boolean;
+        readFile: (path: string, encoding?: string) => string;
+        readDirectory: (path: string, extension?: string, exclude?: string[], depth?: number) => string[];
+        writeFile: (path: string, data: string, writeByteOrderMark?: boolean) => void;
+    };

```

Comment:
Why are you declaring the above interface, which is a subset of `System`, rather than just using `System`? Isn't the type of arg passed a `System` anyway?